### ISSUE TITLE
Allow for no strategy for misfits

### ISSUE THREAD

**Issue by acarril** [manmartgarc/stochatreat#23]
Currently, the user is forced to select a strategy to deal with misfits, which is either "stratum" or "global". It would be nice to have the option to not do anything, and just leave misfits without any assignment (e.g. as `NA`s).

https://github.com/manmartgarc/stochatreat/blob/38d0e8d087c11178a6dd5ecd7d518527153bf945/src/stochatreat/stochatreat.py#L137-L138

I can think of two reasons why this would be useful:
1. It makes it easier to debug/diagnose an allocation setup, as it allows you to see which observations are internally assigned to misfits.
2. It allows the researcher to deal with misfits in potentially different ways from the two pre-specified ones.

I suppose this is easy to implement, but I'd be happy to help if help is needed.

**Comment by manmartgarc** [comment]
Totally agree! Feel free to submit a PR with the option, otherwise I might be able to get around doing this next week.

**Comment by acarril** [comment]
Awesome, I’ll take a look tomorrow and see if I can implement it.

**Comment by acarril** [comment]
Now that I took a closer look at the inner workings of the main method, I think it is quite trivial to identify misfits _ex-post_, as they are stored in the `data` object returned by `stochatreat()`: observations with `stratum_id==np.Inf` are assigned a treatment arm, but are classified as misfits before that.

Unless I am missing something, this means that a user can indeed back out which observations are misfits in the assignment, and therefore the new misfit strategy that I proposed might be unnecessary. Nevertheless, it is not evident that one can do this, and so it might be good to
1. explain this situation more explicitly in the examples and/or docstrings, or
2. implement the option anyway, as I now see it would be fairly simple (e.g. use something like `data.loc[data["stratum_id"] == np.Inf, "treat"] = np.NAN` near the end)

Let me know what you think.

**Comment by manmartgarc** [comment]
Agreed. Let's make sure those are indeed the misfits (via unit tests), and maybe implement a simple method to retrieve the misfits after an assignment to make it explicit for the user that this is how to retrieve them.

Can probably help with this next week.
